pkg/git/remote_git_resolver.go

```diff
@@ -2,59 +2,94 @@ package git
 
 import (
 	gogit "github.com/go-git/go-git/v5"
-	"github.com/go-git/go-git/v5/config"
 	"github.com/go-git/go-git/v5/plumbing"
 	"github.com/go-git/go-git/v5/plumbing/transport"
 	"github.com/go-git/go-git/v5/storage/memory"
-
 	corev1alpha1 "github.com/pivotal/kpack/pkg/apis/core/v1alpha1"
+	"regexp"
+	"strings"
 )
 
-const defaultRemote = "origin"
-
 type remoteGitResolver struct{}
 
 func (*remoteGitResolver) Resolve(auth transport.AuthMethod, sourceConfig corev1alpha1.SourceConfig) (corev1alpha1.ResolvedSourceConfig, error) {
-	remote := gogit.NewRemote(memory.NewStorage(), &config.RemoteConfig{
-		Name: defaultRemote,
-		URLs: []string{sourceConfig.Git.URL},
-	})
+	var resolvedConfig corev1alpha1.ResolvedSourceConfig
 
-	refs, err := remote.List(&gogit.ListOptions{
+	r, err := gogit.Clone(memory.NewStorage(), nil, &gogit.CloneOptions{
```

Comment:
to improve performance can we just do the clone if there is a subpath and do a list if there isn't?

Reply:
I can do that but this does clone a bare repo with the worktree being nil. Is it worth it to add logic to do this? The two flows are pretty different since one uses remote refs and the other uses the actual repository.